src/definitions.jl

```diff
@@ -693,40 +680,57 @@ adjoint_mul(p::Plan, x::AbstractArray) = adjoint_mul(p, x, AdjointStyle(p))
 function adjoint_mul(p::Plan{T}, x::AbstractArray, ::FFTAdjointStyle) where {T}
     dims = fftdims(p)
     N = normalization(T, size(p), dims)
-    return (p \ x) / N
+    pinv = inv(p)
+    # Optimization: when pinv is a ScaledPlan, check if we can avoid a loop over x.
+    # Even if not, ensure that we do only one pass by combining the normalization with the plan.
+    if pinv isa ScaledPlan && pinv.scale == N
+        return pinv.p * x
+    else
+        return (1/N * pinv) * x
```

Comment:
I don't quite follow, could you explain or give an example?

Reply:
Apologies that I still don't follow: why does where we divide by `N` affect the promotion behavior?

Reply:
And why is `N` a `Float64` in the first place, rather than a quantity of type `T`?

Reply:
Oh, I think I understand -- should it be `inv(N)`?

Reply:
```julia
julia> using FFTW

julia> P = plan_fft(rand(Float32, 3, 3));

julia> P' * rand(Float32, 3, 3)
3×3 Matrix{ComplexF32}:
    3.48416+0.0im       -0.264397-0.859624im  -0.264397+0.859624im
 -0.0523138-0.781731im  -0.462729-0.257876im  -0.902054+0.397495im
 -0.0523138+0.781731im  -0.902054-0.397495im  -0.462729+0.257876im
```
Could you provide an example of the issue?

Reply:
Bump

Reply:
Bump -- I still don't follow your initial comment, since as `Int64` and `Float32` promote to `Float32`. An example would be much appreciated.

Reply:
Bump. If this a problem that exists on the main branch, could you make an issue for it, with a reproducer?

Reply:
Sorry, somehow this conversation was lost in my stream of Github notifications.

> And why is N a Float64 in the first place, rather than a quantity of type T?

I assumed it would be an `Int` (and hence `1/N` would be a `Float64`) but you're right of course that it is of type `T`. Maybe one could even use `T = eltype(pinv)` here since the computations are performed with `pinv` but I think it should not matter. Possibly `inv(N)` would be a tiny bit safer since then no `Int`s would be involved but I can't think of any relevant case where it should matter that the inverse is computed in a way that involves an `Int`.